orchestrator/orchestrator/src/ethereum_event_watcher.rs

```diff
@@ -38,13 +38,15 @@ pub async fn check_for_events(
     let our_cosmos_address = our_private_key.to_address(&contact.get_prefix()).unwrap();
 
     let ending_block = if USE_FINALIZATION {
+        // get this first in case inbetween the calls is a block boundary
+        // don't accidentally use this variable elswhere
+        let unsafe_latest_block = get_latest_block_number_with_retry(web3).await;
+
```

Comment:
I moved this upwards because there could be a spurious race condition where a new block is produced or finalized in the middle, this order won't cause spurious failures I don't think